src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1524,10 +1524,8 @@ public sealed override bool IsTupleCompatible(out int tupleCardinality)
             return false;
         }
 
-        // PROTOTYPE: Should be abstract.
-        internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => this;
+        internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => throw ExceptionUtilities.Unreachable;
```

Comment:
>internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => throw ExceptionUtilities.Unreachable; [](start = 8, length = 103)

I think we would want this to be abstract to make sure it is properly implemented by all leaf subclasses. Also, the purpose of this method isn't obvious, could benefit from a doc-comment.